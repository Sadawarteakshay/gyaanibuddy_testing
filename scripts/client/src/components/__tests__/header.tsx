import React from 'react';

import { render, cleanup, screen } from '../../test-utils';
import Header from '../header';
import renderer from 'react-test-renderer';

describe('Header', () => {
  // automatically unmount and cleanup DOM after the test is finished.
  afterEach(cleanup);

  it('renders without error', () => {
    render(<Header />);
  });

  it('render snapshot test', () => {
    const HeaderElement = <Header />;

    render(HeaderElement);

    // Snapshot test - ensure button has no unsuspecting changes between commits
    const ButtonTree = renderer
      .create(HeaderElement)
      .toJSON();

    expect(ButtonTree).toMatchSnapshot();
  });

  it('should contain correct img', () =>{
    const HeaderElement = <Header />;

    render(HeaderElement);

    expect(screen.getByAltText('Space dog')).toHaveAttribute('src', 'dog-3.png');
  });
});
