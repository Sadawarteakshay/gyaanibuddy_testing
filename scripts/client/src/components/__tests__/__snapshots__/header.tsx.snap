// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render snapshot test 1`] = `
<div
  className="css-14estru"
>
  <img
    alt="Space dog"
    className="css-z91ymt"
    src="dog-3.png"
  />
  <div>
    <h2>
      Space Explorer
    </h2>
    <h5
      className="css-1sykydy"
    >
      ée
    </h5>
  </div>
</div>
`;
